hdf/src/atom.c

```diff
@@ -146,10 +143,11 @@ HAinit_group(group_t grp,      /* IN: Group to initialize */
     /* Assertion necessary for faster pointer swapping */
     assert(sizeof(hdf_pint_t) == sizeof(void *));
 
-#ifdef HASH_SIZE_POWER_2
+    /* Ensure hash_size is not zero and a power of two */
+    if (hash_size == 0)
+        HGOTO_ERROR(DFE_ARGS, FAIL);
     if (hash_size & (hash_size - 1))
         HGOTO_ERROR(DFE_ARGS, FAIL);
-#endif /* HASH_SIZE_POWER_2 */
```

Comment:
The original test allowed zero to be a valid size, which is probably not desired